rules/transition_support.bzl

```diff
@@ -112,10 +162,17 @@ def _apple_rule_transition_impl(settings, attr):
         "//command_line_option:tvos_minimum_os": _min_os_version_or_none(attr, attr_platforms, "tvos", platform_type),
         "//command_line_option:watchos_minimum_os": _min_os_version_or_none(attr, attr_platforms, "watchos", platform_type),
     }
+    ret.update(_objc_library_transition_values(settings, platform_type))
     return ret
 
 _supports_visionos = hasattr(apple_common.platform_type, "visionos")
 
+# `--apple_compiler` was removed in https://github.com/bazelbuild/bazel/commit/1acdfc422e724b4fe12c7bf5248086ab514ec4be
+_supports_clo_apple_compiler = _bazel_major_version < 7
```

Comment:
Can we just delete it completely, similarly to: https://github.com/bazelbuild/rules_apple/pull/2040